code/controllers/subsystem/xenoarchaeology.dm

```diff
@@ -198,14 +187,96 @@ SUBSYSTEM_DEF(xenoarchaeology)
 				labeler_tooltip_stats["[initial(T.label_name)]"]["availability"] += list(list("color" = initial(M.material_color), "icon" = initial(M.label_icon)))
 	return temp
 
-/datum/controller/subsystem/xenoarchaeology/proc/build_trait_filters()
-	var/list/temp = list()
-	for(var/datum/xenoartifact_trait/T as() in xenoa_all_traits)
-		T = new T()
-		var/list/hints = T.get_dictionary_hint()
-		for(var/i in hints)
-			if(!temp[i["icon"]])
-				temp[i["icon"]] = list()
-			temp[i["icon"]] += list("[initial(T.label_name)]")
-		QDEL_NULL(T)
-	return temp
+/*
+	Datum for holding a bunch of listed traits for a certain material
+	Thanks, EvilDragon
+*/
+
+/datum/xenoa_material_traits
```

Comment:
My suggestion meant having this feature in `/datum/xenoartifact_material`, so adding a new material will be easier to maintain.
in this case, `/datum/xenoartifact_material` instance should be called from the xeno subsystem. `compile_artifact_whitelist()` proc will not need a parameter.
(We can talk about this change request for more detail later. I am tired now)